### ISSUE TITLE
 Coordinate Addition module & Routing

### ISSUE THREAD

**Issue by S-rz** [benedekrozemberczki/CapsGNN#13]
Hi, thanks for your codes of GapsGNN. And I have some questions about Coordinate Addition module and Routing.
1. Do you use Coordinate Addition module in this codes?
2. In /src/layers.py, line 137 : c_ij = torch.nn.functional.softmax(b_ij, dim=0) . 
    At this time, b_ij.size(0) == 1, why use dim =0 ?

Thanks again.

**Comment by S-rz** [comment]
Hi, I looked at the Capsule Layer codes that you referenced, and the softmax certainly use the wrong dimension. 
The correct call should be:
`c_ij = torch.nn.functional.softmax(b_ij, dim=2)`
And the u_vj1 is too small in the case of using this dataset, so prepossessing should be used, like this:
` b_ij = b_ij + u_vj1`
`b_max = torch.max(b_ij, dim = 2, keepdim = True)`
`b_ij = b_ij / b_max.values` 
After modification, the model has a more reasonable accuracy.

**Comment by benedekrozemberczki** [comment]
Can you open a PR?

**Comment by S-rz** [comment]
@benedekrozemberczki I just changed the for loop in routing algorithm, the softmax of b_ij uses dim = 2 is right, but I do not know how to process the data to make the softmax work correctly.  
As I mentioned before, u_vj1 and b_ij are too small,  computer sees them as zeros, so the softmax of b_ij (dim = 2) is equivalent to ( 1 / b_ij.size(2) ). However, the method I proposed yesterday should be wrong, because the change of b_ij may affect its subsequent update, I do not have a good solution so I did not create a PR.
Sorry, I am a beginner, please let me know if you have a good solution, thanks.

**Comment by JerryFlymi** [comment]
> Hi, I looked at the Capsule Layer codes that you referenced, and the softmax certainly use the wrong dimension.
> The correct call should be:
> `c_ij = torch.nn.functional.softmax(b_ij, dim=2)`
> And the u_vj1 is too small in the case of using this dataset, so prepossessing should be used, like this:
> ` b_ij = b_ij + u_vj1`
> `b_max = torch.max(b_ij, dim = 2, keepdim = True)`
> `b_ij = b_ij / b_max.values`
> After modification, the model has a more reasonable accuracy.

I think so.
